modules/dataquery/php/documents.class.inc

```diff
@@ -0,0 +1,88 @@
+<?php declare(strict_types=1);
+/**
+ * Handles retrieving documents from CouchDB.
+ *
+ * PHP Version 7
+ *
+ * @category Loris
+ * @package  DQT
+ * @author   Alizée Wickenheiser <alizee.wickenheiser@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+namespace LORIS\dataquery;
+
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Message\ResponseInterface;
+
+/**
+ * The Documents class for the GET requests.
+ *
+ * @category Loris
+ * @package  DQT
+ * @author   Alizée Wickenheiser <alizee.wickenheiser@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class Documents extends \NDB_Page
+{
+    public $skipTemplate = true;
+
+    /**
+     * Returns true if the user has permission to access
+     * the DQT module
+     *
+     * @param \User $user The user whose access is being checked
+     *
+     * @return bool true if user has permission
+     */
+    function _hasAccess(\User $user) : bool
+    {
+        return ($user->hasPermission('dataquery_view'));
+    }
+
+    /**
+     * This function will return a json response.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        // Ensure GET or POST request.
+        switch ($request->getMethod()) {
+        case 'GET':
+            return $this->_handleGET($request);
+        default:
+            return new \LORIS\Http\Response\MethodNotAllowed(
+                $this->allowedMethods()
```

Comment:
It doesn't look like this method exists.

Reply:
I'll check this out tomorrow.